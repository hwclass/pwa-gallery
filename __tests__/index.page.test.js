/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Index from '../pages/index.js'

describe('Index page', () => {
  beforeEach(() => {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  })

  it('should render', () => {
    const url = {
      query: {
        photoUri: 'photo-uri'
      }
    }
    const data = {
      title: 'App Title',
      images: [{
        "uri": "some-url/for-image-1",
        "set": "111"
      }, {
        "uri": "some-url/for-image-2",
        "set": "222"
      }, {
        "uri": "some-url/for-image-3",
        "set": "333"
      }, {
        "uri": "some-url/for-image-4",
        "set": "444"
      }, {
        "uri": "some-url/for-image-5",
        "set": "555"
      }]
    }
    const component = shallow(<Index url={url} data={data}/>)
    expect(component.getElements()).toMatchSnapshot();
  })
})

describe('Index page', () => {
  it('should match with snapshot', () => {
    const url = {
      query: {
        photoUri: 'photo-uri'
      }
    }
    const data = {
      title: 'App Title',
      images: [{
        "uri": "some-url/for-image-1",
        "set": "111"
      }, {
        "uri": "some-url/for-image-2",
        "set": "222"
      }, {
        "uri": "some-url/for-image-3",
        "set": "333"
      }, {
        "uri": "some-url/for-image-4",
        "set": "444"
      }, {
        "uri": "some-url/for-image-5",
        "set": "555"
      }]
    }
    const component = renderer.create(<Index url={url} data={data}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})