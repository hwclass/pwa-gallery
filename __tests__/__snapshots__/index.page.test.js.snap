// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match with snapshot 1`] = `
<main
  className="jsx-3097182411 content"
>
  <div
    className="jsx-475590090 frame"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="jsx-475590090 title"
    >
      App Title
      <button
        className="jsx-475590090 closeIcon"
      />
    </span>
    <picture
      className="jsx-475590090 pic"
    >
      <img
        className="jsx-475590090"
        src="http://photo-uri_27.jpg"
      />
    </picture>
  </div>
  <div
    className="jsx-3488773145"
  >
    <header
      className="jsx-1922746274"
    >
      Car Photos
    </header>
    <div
      className="jsx-2977150191 gallery"
    >
      <a
        className="jsx-2977150191 photoLink"
        href="/photo?id=some-url/for-image-1"
        onClick={[Function]}
      >
        <article>
          <picture
            className="jsx-3329984789 pic"
          >
            <source
              className="jsx-3329984789"
              media="(min-width: 36em)"
              sizes="33.3vw"
              srcSet="http://some-url/for-image-1_27.jpg 1024w, http://some-url/for-image-1_27.jpg 640w, http://some-url/for-image-1_27.jpg 320w"
            />
            <source
              className="jsx-3329984789"
              srcSet="http://some-url/for-image-1_27.jpg 2x, http://some-url/for-image-1_2.jpg 1x"
            />
            <img
              className="jsx-3329984789"
              src="http://some-url/for-image-1_27.jpg"
            />
          </picture>
          <span
            className="jsx-2158314212 title"
          >
            App Title
          </span>
        </article>
      </a>
      <a
        className="jsx-2977150191 photoLink"
        href="/photo?id=some-url/for-image-2"
        onClick={[Function]}
      >
        <article>
          <picture
            className="jsx-3329984789 pic"
          >
            <source
              className="jsx-3329984789"
              media="(min-width: 36em)"
              sizes="33.3vw"
              srcSet="http://some-url/for-image-2_27.jpg 1024w, http://some-url/for-image-2_27.jpg 640w, http://some-url/for-image-2_27.jpg 320w"
            />
            <source
              className="jsx-3329984789"
              srcSet="http://some-url/for-image-2_27.jpg 2x, http://some-url/for-image-2_2.jpg 1x"
            />
            <img
              className="jsx-3329984789"
              src="http://some-url/for-image-2_27.jpg"
            />
          </picture>
          <span
            className="jsx-2158314212 title"
          >
            App Title
          </span>
        </article>
      </a>
      <a
        className="jsx-2977150191 photoLink"
        href="/photo?id=some-url/for-image-3"
        onClick={[Function]}
      >
        <article>
          <picture
            className="jsx-3329984789 pic"
          >
            <source
              className="jsx-3329984789"
              media="(min-width: 36em)"
              sizes="33.3vw"
              srcSet="http://some-url/for-image-3_27.jpg 1024w, http://some-url/for-image-3_27.jpg 640w, http://some-url/for-image-3_27.jpg 320w"
            />
            <source
              className="jsx-3329984789"
              srcSet="http://some-url/for-image-3_27.jpg 2x, http://some-url/for-image-3_2.jpg 1x"
            />
            <img
              className="jsx-3329984789"
              src="http://some-url/for-image-3_27.jpg"
            />
          </picture>
          <span
            className="jsx-2158314212 title"
          >
            App Title
          </span>
        </article>
      </a>
      <a
        className="jsx-2977150191 photoLink"
        href="/photo?id=some-url/for-image-4"
        onClick={[Function]}
      >
        <article>
          <picture
            className="jsx-3329984789 pic"
          >
            <source
              className="jsx-3329984789"
              media="(min-width: 36em)"
              sizes="33.3vw"
              srcSet="http://some-url/for-image-4_27.jpg 1024w, http://some-url/for-image-4_27.jpg 640w, http://some-url/for-image-4_27.jpg 320w"
            />
            <source
              className="jsx-3329984789"
              srcSet="http://some-url/for-image-4_27.jpg 2x, http://some-url/for-image-4_2.jpg 1x"
            />
            <img
              className="jsx-3329984789"
              src="http://some-url/for-image-4_27.jpg"
            />
          </picture>
          <span
            className="jsx-2158314212 title"
          >
            App Title
          </span>
        </article>
      </a>
      <a
        className="jsx-2977150191 photoLink"
        href="/photo?id=some-url/for-image-5"
        onClick={[Function]}
      >
        <article>
          <picture
            className="jsx-3329984789 pic"
          >
            <source
              className="jsx-3329984789"
              media="(min-width: 36em)"
              sizes="33.3vw"
              srcSet="http://some-url/for-image-5_27.jpg 1024w, http://some-url/for-image-5_27.jpg 640w, http://some-url/for-image-5_27.jpg 320w"
            />
            <source
              className="jsx-3329984789"
              srcSet="http://some-url/for-image-5_27.jpg 2x, http://some-url/for-image-5_2.jpg 1x"
            />
            <img
              className="jsx-3329984789"
              src="http://some-url/for-image-5_27.jpg"
            />
          </picture>
          <span
            className="jsx-2158314212 title"
          >
            App Title
          </span>
        </article>
      </a>
    </div>
  </div>
</main>
`;

exports[`Index page should render 1`] = `
Array [
  <main
    className="jsx-3097182411 content"
  >
    <_class
      onClose={[Function]}
      title="App Title"
      uri="photo-uri"
    />
    <_class
      images={
        Array [
          Object {
            "set": "111",
            "uri": "some-url/for-image-1",
          },
          Object {
            "set": "222",
            "uri": "some-url/for-image-2",
          },
          Object {
            "set": "333",
            "uri": "some-url/for-image-3",
          },
          Object {
            "set": "444",
            "uri": "some-url/for-image-4",
          },
          Object {
            "set": "555",
            "uri": "some-url/for-image-5",
          },
        ]
      }
      title="App Title"
    />
    <JSXStyle
      css="header.jsx-3097182411{grid-area:main;}"
      styleId="3097182411"
    />
  </main>,
]
`;
